aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+services.AddAntiforgery(options => options.HeaderName = "X-CSRF-TOKEN");
+```
+
+An example of making an AJAX request with the appropriate header, from jQuery:
+<!-- literal_block {"xml:space": "preserve", "language": "javascript", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```javascript
+var csrfToken = $.cookie("CSRF-TOKEN");
+
+$.ajax({
+    url: "/api/password/changepassword",
+    contentType: "application/json",
+    data: JSON.stringify({ "newPassword": "ReallySecurePassword999$$$" }),
+    type: "POST",
+    headers: {
+        "X-CSRF-TOKEN": csrfToken
+    }
+});
+```
+
+## Using IAntiforgery
+
+`IAntiforgery` provides access to the antiforgery system and can be requested in the `Configure` method of the `Startup` class. The following example uses middleware from the app's home page to generate an antiforger token and send it in the response as a cookie (using the default Angular naming convention described above):
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+public void Configure(IApplicationBuilder app, 
+    IAntiforgery antiforgery)
+{
+    app.Use(next => context =>
+    {
+        if (
+            string.Equals(context.Request.Path.Value, "/", StringComparison.OrdinalIgnoreCase) ||
+            string.Equals(context.Request.Path.Value, "/index.html", StringComparison.OrdinalIgnoreCase))
+        {
+            // We can send the request token as a JavaScript-readable cookie, and Angular will use it by default.
+            var tokens = antiforgery.GetAndStoreTokens(context);
+            context.Response.Cookies.Append("XSRF-TOKEN", tokens.RequestToken, 
+                new CookieOptions() { HttpOnly = false });
+        }
+
+        return next(context);
+    });
+    //
+}
+```
+
+### Options
+
+You can customize various options such as the header and cookie names used by antiforgery when you include it in `ConfigureServices`:
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+services.AddAntiforgery(options => 
+{
+  options.CookieDomain = "mydomain.com";
+  options.CookieName = "X-CSRF-TOKEN-COOKIENAME";
+  options.CookiePath = "Path";
+  options.FormFieldName = "AntiforgeryFieldname";
+  options.HeaderName = "X-CSRF-TOKEN-HEADERNAME";
+  options.RequireSsl = false;
+  otpions.SuppressXFrameOptionsHeader = false;
+});
+```
+
```

Comment:
See https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.builder.cookieauthenticationoptions for more info.